nni/algorithms/compression/pytorch/pruning/structured_pruning_masker.py

```diff
@@ -677,13 +683,7 @@ class SlimPrunerMasker(WeightMasker):
 
     def __init__(self, model, pruner, **kwargs):
         super().__init__(model, pruner)
-        weight_list = []
-        for (layer, _) in pruner.get_modules_to_compress():
-            weight_list.append(layer.module.weight.data.abs().clone())
-        all_bn_weights = torch.cat(weight_list)
-        k = int(all_bn_weights.shape[0] * pruner.config_list[0]['sparsity'])
-        self.global_threshold = torch.topk(
-            all_bn_weights.view(-1), k, largest=False)[0].max()
+        self.global_threshold = None
```

Comment:
why we take this part in pruner? I think masker means how to calculate mask and calculate threshold is a part of the mask calculation? Pruner means when and where to calculate mask?

Reply:
Thanks for your suggestion, move the `_get_global_threshold` to masker. Please review the latest version.